flake.nix

```diff
@@ -0,0 +1,128 @@
+{
+  inputs = {
+    crane = {
+      url = "github:ipetkov/crane";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+    nixpkgs.url = "nixpkgs/nixos-unstable";
+    fenix = {
+      url = "github:nix-community/fenix";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+    flake-parts.url = "github:hercules-ci/flake-parts";
+    gitignore = {
+      url = "github:hercules-ci/gitignore";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+  };
+
+  outputs = inputs:
+    inputs.flake-parts.lib.mkFlake {inherit inputs;} {
+      systems = ["x86_64-linux" "aarch64-linux" "aarch64-darwin"];
+      perSystem = {
+        lib,
+        pkgs,
+        inputs',
+        system,
+        ...
+      }: let
+        rustToolchain = inputs'.fenix.packages.stable.toolchain;
+        craneLib = inputs.crane.lib.${system}.overrideToolchain rustToolchain;
+      in {
+        packages = {
+          cargo-pgrx = craneLib.buildPackage {
+            src = inputs.gitignore.lib.gitignoreSource ./.;
+            inherit (craneLib.crateNameFromCargoToml {cargoToml = ./cargo-pgrx/Cargo.toml;}) pname version;
+            cargoExtraArgs = "--package cargo-pgrx";
+            nativeBuildInputs = [pkgs.pkg-config];
+            buildInputs = [pkgs.openssl] ++ lib.optionals pkgs.stdenv.isDarwin [pkgs.Security];
+            # fixes to enable running pgrx tests
+            preCheck = ''
+              export PGRX_HOME=$(mktemp -d)
+            '';
+            # skip tests that require pgrx to be initialized using `cargo pgrx init`
+            cargoTestExtraArgs = "-- --skip=command::schema::tests::test_parse_managed_postmasters";
+          };
+        };
+      };
+
+      flake.lib.buildPgrxExtension = {
+        rustToolchain,
+        system,
+        src,
+        postgresql,
+        additionalFeatures ? [],
+      }: let
+        cargo-pgrx = inputs.self.packages.${system}.cargo-pgrx;
+        pkgs = inputs.nixpkgs.legacyPackages.${system};
+        craneLib = inputs.crane.lib.${system}.overrideToolchain rustToolchain;
+
+        postgresMajor = inputs.nixpkgs.lib.versions.major postgresql.version;
+        cargoToml = builtins.fromTOML (builtins.readFile "${src}/Cargo.toml");
+        name = cargoToml.package.name;
+        pgrxFeatures = builtins.toString additionalFeatures;
+
+        preBuildAndTest = ''
+          export PGRX_HOME=$(mktemp -d)
+          mkdir -p $PGRX_HOME/${postgresMajor}
+
+          cp -r -L ${postgresql}/. $PGRX_HOME/${postgresMajor}/
+          chmod -R ugo+w $PGRX_HOME/${postgresMajor}
+          cp -r -L ${postgresql.lib}/lib/. $PGRX_HOME/${postgresMajor}/lib/
+
+          ${cargo-pgrx}/bin/cargo-pgrx pgrx init \
+            --pg${postgresMajor} $PGRX_HOME/${postgresMajor}/bin/pg_config \
+        '';
+
+        craneCommonBuildArgs = {
+          inherit src;
+          pname = "${name}-pg${postgresMajor}";
+          nativeBuildInputs = [
+            pkgs.pkg-config
+            pkgs.rustPlatform.bindgenHook
```

Comment:
This is a handy package I found that sets `BINDGEN_EXTRA_CLANG_ARGS`, which was previously being done manually in the build step. See https://github.com/symphorien/nixpkgs/blob/master/pkgs/build-support/rust/hooks/rust-bindgen-hook.sh